src/shared/sleep-config.c

```diff
@@ -135,6 +135,59 @@ int read_battery_capacity_percentage(void) {
         return battery_capacity;
 }
 
+static int get_battery_discharge_rate_filepath(char **ret_path) {
+        sd_id128_t machine_id;
+        int r;
+
+        assert(ret_path);
+
+        r = sd_id128_get_machine(&machine_id);
+        if (r < 0)
+                return log_debug_errno(r, "Failed to get machine ID: %m");
+
+        *ret_path = path_join("/var/lib/systemd/sleep", SD_ID128_TO_STRING(machine_id), "battery_discharge_percentage_rate_per_hour");
+        if (!*ret_path)
+                return log_oom_debug();
+        return 0;
+}
+
+int get_battery_discharge_rate(void) {
+        _cleanup_free_ char *filepath = NULL;
+        char *discharge_rate;
+        int stored_discharge_rate, r;
+
+        r = get_battery_discharge_rate_filepath(&filepath);
+        if (r < 0)
+               return r;
+
+        r = read_one_line_file(filepath, &discharge_rate);
+        if (r < 0)
+               return log_debug_errno(r, "Failed to read discharge rate from %s: %m", filepath);
+
+        r = safe_atoi(discharge_rate, &stored_discharge_rate);
+        if (r < 0)
+               return log_debug_errno(r, "Failed to parse discharge rate read from %s location: %m", filepath);
+
+        if (stored_discharge_rate > 0 && stored_discharge_rate < 200)
+               return stored_discharge_rate;
+        return log_debug_errno(SYNTHETIC_ERRNO(EINVAL), "Invalid battery discharge percentage rate per hour: %m");
+}
+
+int put_battery_discharge_rate(int estimated_battery_discharge_rate) {
+        _cleanup_free_ char *filepath = NULL;
```

Comment:
In most of our other code we'd use `p` as the name for a local variable representing a path so I'd do the same here. This is kinda bikeshedding though.

Reply:
Yes. Understood.